src/vm/memory/relocatable.zig

```diff
@@ -887,6 +1018,62 @@ test "MaybeRelocatable: any comparision should return false if other MaybeReloca
     try expect(!maybeRelocatable1.ge(maybeRelocatable2));
 }
 
+test "MaybeRelocatable: add between two relocatable should return a Math error" {
+    try expectError(
+        MathError.RelocatableAdd,
+        fromSegment(0, 10).add(fromSegment(0, 10)),
+    );
+}
+
+test "MaybeRelocatable: add between a Relocatable and a Felt252 should return a proper MaybeRelocatable" {
+    try expectEqual(
+        fromSegment(0, 20),
+        try fromSegment(0, 10).add(fromU256(10)),
+    );
+}
+
+test "MaybeRelocatable: add between two Felt252 should return a proper MaybeRelocatable" {
+    try expectEqual(
+        fromU256(20),
+        try fromU256(10).add(fromU256(10)),
+    );
+}
+
+test "MaybeRelocatable: add between a Felt252 and a Relocatable should return a proper MaybeRelocatable" {
+    try expectEqual(
+        fromSegment(0, 20),
+        try fromU256(10).add(fromSegment(0, 10)),
+    );
+}
+
+test "MaybeRelocatable: sub between two Relocatable should return a proper MaybeRelocatable" {
```

Comment:
extra test for `sub` between two Relocatables with different segment index?

Reply:
@lana-shanghai Done :)